python/unitytrainers/models.py

```diff
@@ -190,12 +190,12 @@ def create_dc_actor_critic(self, h_size, num_layers):
         self.output = tf.identity(self.output, name="action")
         self.value = tf.layers.dense(hidden, 1, activation=None)
         self.value = tf.identity(self.value, name="value_estimate")
-        self.entropy = -tf.reduce_sum(self.probs * tf.log(self.probs + 1e-10), axis=1)
-        self.action_holder = tf.placeholder(shape=[None], dtype=tf.int32)
+        self.entropy = -tf.reduce_sum(self.all_probs * tf.log(self.all_probs + 1e-10), axis=1)
+        self.action_holder = tf.placeholder(shape=[None], dtype=tf.int32, name="action_input")
```

Comment:
GridWorld does not work for me because action_holder is not supplied when running take_action

Reply:
GridWorld trains fine for me. `action_holder` is only necessary when training, not when collecting experiences in `take_action`. What is the specific error you are getting?